vscode-extension/editor/src/VSCodeEditor.tsx

```diff
@@ -0,0 +1,449 @@
+import {
+  AIConfigEditor,
+  AIConfigCallbacks,
+  RunPromptStreamCallback,
+  RunPromptStreamErrorCallback,
+  RunPromptStreamErrorEvent,
+  LogEvent,
+  LogEventData,
+} from "@lastmileai/aiconfig-editor";
+import { Flex, Loader, Image, createStyles } from "@mantine/core";
+import {
+  AIConfig,
+  InferenceSettings,
+  JSONObject,
+  Output,
+  Prompt,
+} from "aiconfig";
+import yaml from "js-yaml";
+import { useCallback, useContext, useEffect, useMemo, useState } from "react";
+import { ufetch } from "ufetch";
+import { datadogLogs } from "@datadog/browser-logs";
+import WebviewContext from "./WebviewContext";
+import { ROUTE_TABLE } from "./utils/api";
+import { streamingApiChain } from "./utils/oboeHelpers";
+import {
+  getWebviewState,
+  notifyDocumentDirty,
+  updateWebviewState,
+} from "./utils/vscodeUtils";
+
+const useStyles = createStyles(() => ({
+  editorBackground: {
+    margin: "0 auto",
+    minHeight: "100vh",
+  },
+}));
+
+const MODE = "vscode";
+
+export default function VSCodeEditor() {
+  const { vscode } = useContext(WebviewContext);
+
+  // TODO: saqadri - does this need to be wrapped in a memo?
+  const webviewState = getWebviewState(vscode);
```

Comment:
In the current code it doesn't since it's only used in one place (initializing the state). It would actually be better to just call it within the state initialization instead of here (where it will be called unnecessarily each render). But maybe it's used elsewhere later in the stack.